src/app/components_ngrx/work-item-link/work-item-link.component.ts

```diff
@@ -231,6 +235,20 @@ export class WorkItemLinkComponent implements OnInit {
     }));
   }
 
+  createRouterLink(wiNumber, detailContext) {
+    if (detailContext === "detail") {
+      let url = this.router.url.split('detail');
+      url[url.length-1] = 'detail/' + wiNumber;
+      return url.join("");
+    }
```

Comment:
@vikram-raj there was no need of creating this function, you just had to add this `../` here `<a [routerLink]=" ' ../' + wi.number" *ngIf="detailContext=='detail' ">` as `../` this can be used to go to the higher level and just add the workitem number there.